traits/tests/test_trait_list_object.py

```diff
@@ -112,28 +112,28 @@ def test_init(self):
         tl = TraitList([1, 2, 3])
 
         self.assertListEqual(tl, [1, 2, 3])
-        self.assertIs(tl.item_validator, accept_anything)
+        self.assertIs(tl.item_validator, validate_everything)
```

Comment:
This is off-topic for this PR, but for what it's worth, I think this is the wrong thing to be testing (and that's likely my fault - the test code evolved through several stages): instead of testing that the `item_validator` _is_ one particular function, we should instead test the behaviour: that without an explicitly given item validator, everything is accepted as valid. Then we'd be able to drop the `validate_everything` import entirely in this test module.

Reply:
That seems to creep more towards integrations tests rather than unit tests. Why would it be preferable over the current approach?

To me it makes more sense to test common "everything is accepted" behaviour in unit tests of `validate_everything` (which don't exist at the moment, but can be added in this PR) and then only checking for the right validator function in unit tests of `TraitList`, `TraitSet` and `TraitDict` with the assumption that `validate_everything` works as expected.

Reply:
Mostly because the fact that we have `validate_everything` there is an implementation detail, while the purpose of the tests (unit tests or otherwise) is to test behaviour. We should be at liberty to replace the default item validator with a created-at-need `lambda` function, or just leave it as `None`, without disrupting the tests. The general principle here is that if you bind the tests too closely to the implementation, you end up making code unnecessarily hard to change or refactor.

Reply:
Okay, that makes sense. Thanks for the explanation!

I'll leave the tests alone in this PR and open an issue to fix them later

Reply:
A nuance here is that we _document_ the `item_validator` attribute under the `Attributes` section, so we're effectively guaranteeing that the user can retrieve _something_ that behaves like an item validator by doing `my_list_trait_thing.item_validator`. So we can't just leave it as `None`, but we don't owe the user anything more than _some_ callable that has the right effect. There's no benefit in always providing the same callable; the fact that we do so is just a convenience for the developer.

Reply:
Opened https://github.com/enthought/traits/issues/1042

Reply:
Thank you!